lib/chef/provider/apt_repository.rb

```diff
@@ -200,13 +201,15 @@ def key_type(uri)
       def install_key_from_uri(key)
         key_name = key.gsub(/[^0-9A-Za-z\-]/, "_")
         cached_keyfile = ::File.join(Chef::Config[:file_cache_path], key_name)
+        tmp_dir = Dir.mktmpdir(".gpg")
+        at_exit { FileUtils.remove_entry(tmp_dir) }
```

Comment:
you can probably just use `@tmp_dir` here and by assigning it to an instance variable it lives for the lifetime of the provider, which allows the converge phase to complete before the tmpdir gets removed?

oh except if you don't give mktmpdir a block it never deletes it because it only returns a string and not some object that you it could setup a destructor on...

i still feel like `at_exit {}` is kinda weird.

maybe use `ruby_block` to lazy `FileUtils.rm_rf` or something?  except that updates the outer resource and will fire every time...

okay, i think i see why you're doing this...

think i see why phil wanted a ruby_block that didn't update the outer resource and fire off notifications as well...